lib/src/signed_video_tlv.c

```diff
@@ -902,6 +902,42 @@ tlv_find_tag(const uint8_t *tlv_data, size_t tlv_data_size, sv_tlv_tag_t tag, bo
   return NULL;
 }
 
+bool
+tlv_find_and_decode_recurrent_tags(signed_video_t *self,
+    const uint8_t *tlv_data,
+    size_t tlv_data_size)
+{
+  const uint8_t *tlv_data_ptr = tlv_data;
+
+  if (!self || !tlv_data || tlv_data_size == 0) return false;
+
+  svi_rc status = SVI_UNKNOWN;
+  bool recurrent_tags_found = false;
+  while (tlv_data_ptr < tlv_data + tlv_data_size) {
+    size_t tlv_header_size = 0;
+    size_t length = 0;
+    sv_tlv_tag_t this_tag = UNDEFINED_TAG;
+    status = decode_tlv_header(tlv_data_ptr, &tlv_header_size, &this_tag, &length);
+    if (status != SVI_OK) {
+      DEBUG_LOG("Could not decode tlv header");
+      break;
+    }
+    tlv_data_ptr += tlv_header_size;
+    if (!tlv_tuples[this_tag].is_always_present) {
+      recurrent_tags_found = true;
```

Comment:
Setting `recurrent_tags_found` should be done after the status check, that is, after line 933. Otherwise we will mark as `true` even if we failed decoding it.